pkg/cloud/externalconn/record.go

```diff
@@ -89,15 +89,13 @@ func (e *externalConnectionNotFoundError) Error() string {
 // `system.external_connections` table and returns the read-only interface for
 // interacting with it.
 func LoadExternalConnection(
-	ctx context.Context,
-	name string,
-	connectionType connectionpb.ConnectionType,
-	ex sqlutil.InternalExecutor,
-	user username.SQLUsername,
-	txn *kv.Txn,
+	ctx context.Context, name string, ex sqlutil.InternalExecutor, txn *kv.Txn,
 ) (ExternalConnection, error) {
+	// Loading an External Connection is only allowed for users with the `USAGE`
+	// privilege. We run the query as `node` since the user might not have
+	// `SELECT` on the system table.
```

Comment:
Is it normal to be able to use a resource you can't read? Or I guess that's the point of external connections, isn't it?

Reply:
The `SELECT` on the system table is an implementation detail rather than a user inputted query and so it seems appropriate to run it as a "node" user that is capable of querying system tables. The fine-grained privileges are meant to allow non-admin users (can't read system tables) to use these connections if they have the `USAGE` privilege.

Reply:
Thanks for the quick review, I'm going to wait on @RichardJCai for the privileges stuff.